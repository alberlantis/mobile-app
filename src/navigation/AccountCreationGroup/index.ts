export { default } from "./AccountCreationGroup";
export type { AccountCreationParamList } from "./AccountCreationGroup";
