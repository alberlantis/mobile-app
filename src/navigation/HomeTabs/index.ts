export { default } from "./HomeTabs";
export type { HomeTabsParamList } from "./HomeTabs";
