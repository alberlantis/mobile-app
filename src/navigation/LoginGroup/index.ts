export { default } from "./LoginGroup";
export type { LoginGroupParamList } from "./LoginGroup";
