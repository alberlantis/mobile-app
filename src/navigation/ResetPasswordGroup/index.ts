export { default } from "./ResetPasswordGroup";
export type { ResetPasswordGroupParamList } from "./ResetPasswordGroup";
