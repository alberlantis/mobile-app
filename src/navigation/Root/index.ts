export { default } from "./Root";
export type {
  RootScreenProps,
  RootNavigationProps,
  RootRouteProps,
  HomeTabsParams,
} from "./Root";
