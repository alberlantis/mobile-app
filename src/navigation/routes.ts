export const SCREENS = {
  HOME_TABS: "HomeTabs",
  HOME: "Home",
  NOTIFICATIONS: "Notifications",
  ONBOARDING: "Onboarding",
  POSTING: "Posting",
  PROFILE_HOME: "ProfileHome",
  SPLASH: "Splash",
  SIGN_UP: "SignUp",
  NOSTR_UP: "NostrUp",
  LOGIN: "Login",
  NOSTR_IN: "NostrIn",
  INTERESTS: "Interests",
  COMPLETE_PROFILE: "CompleteProfile",
  RECOVERY_EMAIL: "RecoveryEmail",
  RESET_PASSWORD: "ResetPassword",
} as const;
