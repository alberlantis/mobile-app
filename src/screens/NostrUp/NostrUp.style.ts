import { StyleProp, StyleSheet, ViewStyle, Dimensions } from "react-native";

import fonts from "src/theme/fonts";
import colors from "src/theme/colors";

const screenHeight = Dimensions.get("window").height;

export const copyIconSize = fonts[16];
export const getButtonContainer = (showKey: boolean): StyleProp<ViewStyle> => ({
  alignItems: "center",
  width: "100%",
  height: showKey ? "17%" : "52%",
  justifyContent: showKey ? "flex-end" : "flex-start",
});
export default StyleSheet.create({
  container: {
    marginHorizontal: "2%",
    alignItems: "center",
    flex: 1,
  },
  logoContainer: {
    alignItems: "center",
    width: "100%",
    height: "25%",
  },
  inputContainer: {
    alignItems: "center",
    width: "100%",
    height: "15%",
  },
  privateKeyContainer: {
    alignItems: "center",
    width: "100%",
    height: "35%",
    justifyContent: "space-around",
  },
  button: {
    marginBottom: screenHeight * 0.02,
  },
  bottomContainer: {
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
    height: "2%",
  },
  textDescription: {
    color: colors.WHITE_BOLD,
    textAlign: "center",
    fontSize: fonts[14],
  },
  nsecKeyContainer: {
    width: "100%",
    height: "65%",
    backgroundColor: colors.GRAY_LIGHT,
    justifyContent: "space-around",
    alignItems: "center",
    borderRadius: 30,
    paddingHorizontal: "10%",
  },
  nsecTitle: {
    color: colors.WHITE,
    fontWeight: "bold",
    fontSize: fonts[14],
  },
  nsecKey: {
    color: colors.WHITE_BOLD,
    textAlign: "center",
    fontSize: fonts[14],
  },
  copyButton: { color: colors.WHITE_BOLD, marginLeft: 5 },
});
