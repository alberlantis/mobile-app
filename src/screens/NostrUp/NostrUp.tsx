import React, { useState } from "react";

import {
  DefaultBackground,
  Input,
  LogoTitle,
  Button,
} from "src/shared/components";
import { SCREENS } from "src/navigation/routes";
import type { RootScreenProps } from "src/navigation/Root";
import s from "./NostrUp.style";

const NostrUp = ({ navigation }: RootScreenProps<"NostrUp">) => {
  const [username, setUsername] = useState("");

  return (
    <DefaultBackground style={s.container} blurPos="top">
      <LogoTitle title="Sign Up" />
      <Input
        type="username"
        placeholder="Enter your username"
        label="Username"
        onChangeText={setUsername}
        value={username}
        marginTop={20}
      />
      <Button
        marginTop={20}
        onPress={() => navigation.navigate(SCREENS.HOME_TABS)}
        text="Next"
        theme="primary"
        size="extra-large"
      />
    </DefaultBackground>
  );
};

export default NostrUp;
