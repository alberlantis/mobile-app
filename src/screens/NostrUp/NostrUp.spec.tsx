import { fireEvent, render, screen } from "@testing-library/react-native";

import { SCREENS } from "src/navigation/routes";
import NostrUp from "./NostrUp";

jest.mock("src/shared/components", () => {
  const { Text } =
    jest.requireActual<typeof import("react-native")>("react-native");
  return {
    DefaultBackground: ({ children }: { children: React.ReactNode }) => (
      <>{children}</>
    ),
    Button: ({ text, onPress }: { text: string; onPress(): void }) => (
      <Text onPress={onPress}>{`${text}-button`}</Text>
    ),
    LogoTitle: ({ title }: { title: string }) => (
      <Text>{`${title}-title`}</Text>
    ),
    Input: ({ placeholder, label }: { placeholder: string; label: string }) => (
      <>
        {!!label && <Text>{`${label}-label`}</Text>}
        <Text>{`${placeholder}-placeholder`}</Text>
      </>
    ),
  };
});

const mockNavigation: any = {
  navigate: jest.fn(),
};
const mockRoute: any = {
  name: "NostrUp",
};

describe("NostrUp", () => {
  beforeEach(() => {
    render(<NostrUp navigation={mockNavigation} route={mockRoute} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should match snapshot", () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("should render logo title", () => {
    expect(screen.getByText("Sign Up-title")).toBeOnTheScreen();
  });

  it("should render username input with label", () => {
    expect(screen.getByText("Username-label")).toBeOnTheScreen();
    expect(
      screen.getByText("Enter your username-placeholder"),
    ).toBeOnTheScreen();
  });

  it("should render Next button", () => {
    expect(screen.getByText("Next-button")).toBeOnTheScreen();
  });

  it("should navigate to HomeTabs when tapped", () => {
    const button = screen.getByText("Next-button");
    fireEvent.press(button);
    expect(mockNavigation.navigate).toHaveBeenCalledTimes(1);
    expect(mockNavigation.navigate).toHaveBeenCalledWith(SCREENS.HOME_TABS);
  });
});
