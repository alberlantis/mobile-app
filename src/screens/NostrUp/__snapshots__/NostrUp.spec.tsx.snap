// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NostrUp should match snapshot 1`] = `
[
  <Text>
    Sign Up-title
  </Text>,
  <Text>
    Username-label
  </Text>,
  <Text>
    Enter your username-placeholder
  </Text>,
  <Text
    onPress={[Function]}
  >
    Next-button
  </Text>,
]
`;
